pkg/pluginmanager/essentials.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package pluginmanager
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/log"
+)
+
+// getEssentialsPluginGroupDetails is a function that retrieves the name and version of the essentials plugin group.
+func getEssentialsPluginGroupDetails() (name, version string) {
+	// Set the default name for the essential plugin group.
+	name = constants.DefaultCLIEssentialsPluginGroupName
+
+	// Check if the environment variable for the essentials plugin group name is set.
+	// If it is, override the default name with the value from the environment variable.
+	essentialsPluginGroupName := os.Getenv(constants.TanzuCLIEssentialsPluginGroupName)
+	if essentialsPluginGroupName != "" {
+		name = essentialsPluginGroupName
+	}
+
+	// Check if the environment variable for the essentials plugin group version is set.
+	// If it is, set the version to the value from the environment variable.
+	essentialsPluginGroupVersion := os.Getenv(constants.TanzuCLIEssentialsPluginGroupVersion)
+	if essentialsPluginGroupVersion != "" {
+		version = essentialsPluginGroupVersion
+	}
+
+	// Return the name and version of the essentials plugin group.
+	return name, version
+}
+
+// InstallEssentialPluginGroups is a function that installs or upgrades the essential plugin groups.
+func InstallEssentialPluginGroups() (string, error) {
```

Comment:
There is nothing like installing a plugin group. But we are installing plugins from the plugin-group. So to avoid confusion for someone new, how about renaming this function?


```suggestion
func InstallPluginsFromEssentialPluginGroup() (string, error) {
```

Reply:
Updated the method names